ocrd_page_to_alto/styles.py

```diff
@@ -122,7 +122,7 @@ def __init__(self):
     def set_alto_tag_from_type(self, reg_alto, reg_page):
         typ = reg_page.get_type() if hasattr(reg_page, 'get_type') else None
         if typ:
-            if hasattr(reg_alto, 'TYPE'):
+            if hasattr(reg_page, 'type_'):
```

Comment:
This check is there to prevent setting `TYPE` for an ALTO element that doesn't support it. That's most types, only `ComposedBlock` and `Illustration` allow it. Your check verifies that the source region in PAGE has a `type_` attribute - which they all do, so it fixes the issue for your data but produces invalid ALTO.

Have a look at #18 whether that fixes your problem as well. Feel free to share test data / conditions so we can make a regression test for this.

Reply:
To be honest, my intention was just to fix the red badge on the project's page.

The whitelist approach fits far better in this context, I'm fine with it